src/renderer/store/battle.store.ts

```diff
@@ -185,50 +254,64 @@ function getMaxPlayersPerTeam() {
 function updateTeams() {
     if (!battleStore.battleOptions.map) return;
     const numberOfTeams = getNumberOfTeams();
-    const maxPlayersPerTeam = getMaxPlayersPerTeam();
-
-    // Try to keep players on the same team, then fill the other teams, then move extra players to spectators
 
     // Adjust number of teams
-    let extraParticipants: Array<Player | Bot> = [];
     if (battleStore.teams.length < numberOfTeams) {
-        const teamsToAdd = numberOfTeams - battleStore.teams.length;
-        battleStore.teams = [...battleStore.teams, ...new Array(teamsToAdd).fill([])];
+        for (let i = 0; i <= numberOfTeams - battleStore.teams.length; i++) addTeam();
     } else if (battleStore.teams.length > numberOfTeams) {
-        extraParticipants = [...extraParticipants, ...battleStore.teams.slice(numberOfTeams).flat()];
-        battleStore.teams = [...battleStore.teams.slice(0, numberOfTeams)];
+        for (let i = battleStore.teams.length - 1; i + 1 > numberOfTeams; i--) removeTeam(i);
     }
 
-    // Remove extra players/bots from teams
-    for (const [index, team] of battleStore.teams.entries()) {
-        const raptorsOrScavengers = team.find((participant) => isBot(participant) && (isRaptor(participant) || isScavenger(participant)));
-        if (raptorsOrScavengers) {
-            extraParticipants.push(...team.filter((participant) => !isBot(participant) || (!isRaptor(participant) && !isScavenger(participant))));
-            battleStore.teams[index] = [raptorsOrScavengers];
-        } else if (index >= numberOfTeams) {
-            extraParticipants.push(...team);
-        } else if (team.length > maxPlayersPerTeam) {
-            extraParticipants = [...extraParticipants, ...team.slice(maxPlayersPerTeam)];
-            team.splice(0, maxPlayersPerTeam);
+    // Adjust number of participants per team
+    const maxPartipantsPerTeam = getMaxPlayersPerTeam();
+    const extraParticipants: Array<Player | Bot> = [];
+    for (const team of battleStore.teams.values()) {
+        if (team.participants.filter(isScavengerOrRaptor).length > 0) continue;
+
+        if (team.participants.length > maxPartipantsPerTeam) {
+            extraParticipants.push(...team.participants.splice(maxPartipantsPerTeam));
+        } else {
+            team.participants.push(...extraParticipants.splice(0, maxPartipantsPerTeam - team.participants.length));
         }
     }
```

Comment:
maybe here there should be additional loop that calls remove functions for remaining extra participants, but that would be mostly defensive programming, as it stands all works fine.